plugins/CoreUpdater/SystemSettings.php

```diff
@@ -54,6 +59,13 @@ protected function init()
         $isWritable = $isWritable && PluginUpdateCommunication::canBeEnabled();
         $this->sendPluginUpdateEmail = $this->createSendPluginUpdateEmail();
         $this->sendPluginUpdateEmail->setIsWritableByCurrentUser($isWritable);
+
+        $isWritable = Piwik::hasUserSuperUserAccess() && CoreAdminController::isGeneralSettingsAdminEnabled();
+        $dbSettings = new Settings();
+        if ($dbSettings->getUsedCharset() !== 'utf8mb4' && DbHelper::getDefaultCharset() === 'utf8mb4') {
```

Comment:
just fyi was worried this might issue DB requests during tracking but double checked looks like these system settings aren't used during tracking 👍  Might be good though to change the `if` to `if ($isWritable && $dbSettings->getUsedCharset() ...)`? This way eg on cloud we avoid these queries there.

Reply:
This should be last thing. feel free to merge if/once tests pass